File: app/src/main/java/agersant/polaris/features/browse/BrowseItemHolderAlbumTrack.java
package agersant.polaris.features.browse;

import android.view.View;
import android.widget.TextView;

import java.util.Locale;

import agersant.polaris.CollectionItem;
import agersant.polaris.PlaybackQueue;
import agersant.polaris.R;
import agersant.polaris.Song;
import agersant.polaris.api.API;


class BrowseItemHolderAlbumTrack extends BrowseItemHolder {

    private final TextView trackNumberText;
    private final TextView titleText;

    BrowseItemHolderAlbumTrack(API api, PlaybackQueue playbackQueue, BrowseAdapter adapter, View itemView, View itemQueueStatusView) {
        super(api, playbackQueue, adapter, itemView, itemQueueStatusView);
        trackNumberText = itemView.findViewById(R.id.track_number);
        titleText = itemView.findViewById(R.id.title);
    }

    @Override
    void bindItem(CollectionItem item) {
        super.bindItem(item);

        var song = (Song) item;

        String title = song.getTitle();
        if (title != null) {
            titleText.setText(title);
        } else {
            titleText.setText(song.getName());
        }

        int trackNumber = song.getTrackNumber();
        if (trackNumber >= 0) {
            trackNumberText.setText(String.format((Locale) null, "%1$02d.", trackNumber));
        } else {
            trackNumberText.setText("");
        }
    }

}

File: app/src/main/java/agersant/polaris/features/browse/BrowseItemHolderAlbumTrack.kt
package agersant.polaris.features.browse

import agersant.polaris.CollectionItem
import agersant.polaris.PlaybackQueue
import agersant.polaris.R
import agersant.polaris.api.API
import agersant.polaris.util.formatDuration
import agersant.polaris.util.formatTime
import android.view.View
import android.widget.TextView

internal class BrowseItemHolderAlbumTrack(
    api: API?,
    playbackQueue: PlaybackQueue?,
    adapter: BrowseAdapter?,
    itemView: View,
    itemQueueStatusView: View?
) : BrowseItemHolder(api, playbackQueue, adapter, itemView, itemQueueStatusView) {

    private val trackNumberText: TextView = itemView.findViewById(R.id.track_number)
    private val titleText: TextView = itemView.findViewById(R.id.title)
    private val artistText: TextView = itemView.findViewById(R.id.artist)
    private val durationText: TextView = itemView.findViewById(R.id.duration)

    public override fun bindItem(item: CollectionItem) {
        super.bindItem(item)

        val trackNumber = item.trackNumber
        if (trackNumber != -1) {
            trackNumberText.text = String.format("%02d", trackNumber)
        } else {
            trackNumberText.text = ""
        }
        titleText.text = item.title ?: item.name
        artistText.text = item.artist.orEmpty()
        durationText.text = formatDuration(item.duration)
    }
}
